src/main/java/de/tum/in/www1/artemis/web/rest/FileUploadSubmissionResource.java

```diff
@@ -311,16 +292,11 @@ public FileUploadSubmissionResource(SubmissionRepository submissionRepository, R
         }
 
         Optional<Submission> optionalSubmission = participation.findLatestSubmission();
-        FileUploadSubmission fileUploadSubmission;
-        if (optionalSubmission.isEmpty()) {
-            // this should never happen as the submission is initialized along with the participation when the exercise is started
-            fileUploadSubmission = new FileUploadSubmission();
-            fileUploadSubmission.setParticipation(participation);
-        }
-        else {
-            // only try to get and set the file upload if the fileUploadSubmission existed before
-            fileUploadSubmission = (FileUploadSubmission) optionalSubmission.get();
-        }
+        FileUploadSubmission fileUploadSubmission = (FileUploadSubmission) optionalSubmission.orElseGet(() -> {
```

Comment:
I find it a bit confusing, that an `optionalSubmission` Object is first created and transformed via `tempParticipation` into a `fileUploadSubmission`? Why is it called `tempParticipation` if it's of type `FileUploadSubmission`? Isn't there a more readable way of doing this? I think one could also add a short comment above.